BasicClasses/BasicClasses.tex

```diff
@@ -805,6 +805,62 @@ \section{Booleans}
 \dothis{Try to imagine how \ct{and:} and \ct{or:} are implemented.
     Check the implementations in \ct{Boolean}, \ct{True}, and \ct{False}.}
 
+%=================================================================
+\section{Exceptions}
+
+\ct{Exceptions}\clsindexmain{Exception} might not be the first class you will need when starting to explore \sq. 
+However, \ct{Exceptions} provide a fundamental way to influence the control flow in your application, thus we already introduce them with the other basic classes.
+
+Similar to \ct{ifTrue:} and \ct{ifFalse:}, exceptions are not part of the language, but implemented as ordinary objects in the class library. 
+Thus, we can work with them as we would work with any other object.
+
+The common superclass of all exceptions is the abstract class \ct{Exception}.
+There are numerous subclasses of \ct{Exception} representing specific exceptions, such as \ct{ZeroDivide}, \ct{MessageNotUnderstood}, or \ct{ProgressNotification}.
+In particular, subclasses of \ct{Error} represent erroneous situations, subclasses of \ct{Notification} indicate that something interesting occurred.
+
+To raise an exception you can use the method \mthind{Exception}{signal}.
+If you want to pass a message along to help users figure out what happened, you can use \ct{Exception>>>signal:}.
+When an exception is not handled by anyone, a debugger will open.
+
+\begin{code}{NB: CANNOT TEST}
+Error signal. --> "Will open a debugger"
+Error signal: 'We wanted this to happen'. --> "Will open a debugger with a message"
+\end{code}
+
+As we do not want to present users with debuggers every time something unexpected happens, we can also handle exceptions directly within our code.
+The methods for handling exceptions are associated with blocks, so you can find them in the \ct{exceptions} protocol of the class \ct{BlockClosure}.
+To catch an exception you can use the method \ct{BlockClosure>>>on: anExceptionOrExceptionSet do: aBlock}\cmindex{BlockClosure}{on:do:}.
```

Comment:
Would it be worth noting that `#on:do:` is the equivalent of `try ... catch` in C dialects or `try ... except` in Python, and that `#ensure:` works like `finally` in other languages?